From 788aca8e2665ca9f2a00de5a2dcf2c0357b9b023 Mon Sep 17 00:00:00 2001
From: Michael Haubenwallner <michael.haubenwallner@ssi-schaefer.com>
Date: Wed, 22 May 2019 14:36:12 +0200
Subject: [PATCH 03/19] fix uninitialized discriminator values

Plain old data types lack a constructor for default initialization, and
valgrind does detect them uninitialized in the idl compiler for example.
---
 include/mico/security/csi.h | 4 ++--
 orb/security/csi.cc         | 6 ++++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/mico/security/csi.h b/include/mico/security/csi.h
index ab66c69..7b9cfec 100644
--- a/include/mico/security/csi.h
+++ b/include/mico/security/csi.h
@@ -154,8 +154,8 @@ class IdentityToken {
     } _m;
 
   public:
-    #ifdef HAVE_EXPLICIT_STRUCT_OPS
     IdentityToken();
+    #ifdef HAVE_EXPLICIT_STRUCT_OPS
     ~IdentityToken();
     IdentityToken( const IdentityToken& s );
     IdentityToken& operator=( const IdentityToken& s );
@@ -304,8 +304,8 @@ class SASContextBody {
     } _m;
 
   public:
-    #ifdef HAVE_EXPLICIT_STRUCT_OPS
     SASContextBody();
+    #ifdef HAVE_EXPLICIT_STRUCT_OPS
     ~SASContextBody();
     SASContextBody( const SASContextBody& s );
     SASContextBody& operator=( const SASContextBody& s );
diff --git a/orb/security/csi.cc b/orb/security/csi.cc
index ce3d92e..700da59 100644
--- a/orb/security/csi.cc
+++ b/orb/security/csi.cc
@@ -193,11 +193,12 @@ namespace CSI
 CORBA::TypeCodeConst _tc_IdentityToken;
 }
 
-#ifdef HAVE_EXPLICIT_STRUCT_OPS
 CSI::IdentityToken::IdentityToken()
+	: _discriminator(ITTAbsent)
 {
 }
 
+#ifdef HAVE_EXPLICIT_STRUCT_OPS
 CSI::IdentityToken::IdentityToken( const IdentityToken&_u )
 {
   _discriminator = _u._discriminator;
@@ -904,11 +905,12 @@ namespace CSI
 CORBA::TypeCodeConst _tc_SASContextBody;
 }
 
-#ifdef HAVE_EXPLICIT_STRUCT_OPS
 CSI::SASContextBody::SASContextBody()
+	: _discriminator(MTEstablishContext)
 {
 }
 
+#ifdef HAVE_EXPLICIT_STRUCT_OPS
 CSI::SASContextBody::SASContextBody( const SASContextBody&_u )
 {
   _discriminator = _u._discriminator;
-- 
2.19.2

